frontend/src/GameGrid/hooks/useStoreGrid.ts

```diff
@@ -0,0 +1,30 @@
+import {useState} from 'react';
+import grid from "../Grid";
+
+export const useStoreGrid = () => {
+    const [response, setResponse] = useState(null);
+    const [error, setError] = useState<Error | null>(null);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const postData = async (url: string, data: any) => {
```

Comment:
Just to keep in mind for further development: To handle the potential issue of trying to update the state of an unmounted component in your custom hook useStoreGrid, you can use a cleanup mechanism within your useEffect hook. This mechanism involves tracking the mounted state of the component and updating the state only if the component is still mounted.

```
    const postData = async (url: string, data: any) => {
        let isMounted = true;  // Track the mounted state
        setIsLoading(true);

        try {
            const response = await fetch(url, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ grid: data }),
            });
            const result = await response.json();

            if (isMounted) {
                setResponse(result);
            }
        } catch (err) {
            if (isMounted) {
                setError(err);
            }
        } finally {
            if (isMounted) {
                setIsLoading(false);
            }
        }

        // Cleanup function to set isMounted to false when the component unmounts
        return () => {
            isMounted = false;
        };
    };

    return { postData, response, error, isLoading };
};
```